eva-accession-clustering-automation/cluster_one_assembly.py

```diff
@@ -0,0 +1,130 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import os
+import sys
+import datetime
+import logging
+from create_clustering_properties import create_properties_file
+from create_clustering_properties import check_valid_sources
+from create_clustering_properties import check_vcf_source_requirements
+from config_custom import get_args_from_private_config_file
+from ebi_eva_common_pyutils.command_utils import run_command_with_output
+
+logger = logging.getLogger(__name__)
+timestamp = datetime.datetime.now().strftime("%Y%m%d%H%M%S")
+
+
+def generate_bsub_command(assembly_accession, properties_path, clustering_artifact, automation_timestamp):
+    job_name = 'cluster_' + assembly_accession
+    log_file = assembly_accession + '_cluster_' + timestamp + '.log'
+    error_file = assembly_accession + '_cluster_' + timestamp + '.err'
+
+    command = 'bsub -J ' + job_name + ' -o ' + log_file + ' -e ' + error_file + ' -M 8192 -R "rusage[mem=8192]" ' + \
+              'java -jar ' + clustering_artifact + ' -Dspring.config.location=' + properties_path
+
+    print(command)
+    add_to_command_file(properties_path, command, automation_timestamp)
+    return command
+
+
+def add_to_command_file(properties_path, command, automation_timestamp):
+    """
+    This method writes the commands to a text file in the output folder. If it was run using the
+    cluster_multiple_assemblies method the automation timestamp will be provided and used. That way the commands will
+    be on the same file when ran from the automation script.
+    """
+    commands_path = os.path.dirname(properties_path) + '/commands_'
+    if automation_timestamp:
+        commands_path += automation_timestamp
+    else:
+        commands_path +=  timestamp
+    commands_path += '.txt'
+
+    with open(commands_path, 'a+') as commands:
+        commands.write(command + '\n')
+
+
+def run_clustering(source, vcf_file, project_accession, assembly_accession, private_config_file,
+                   private_config_xml_file, profile, output_directory, clustering_artifact, only_printing,
+                   automation_timestamp):
+    preliminary_check(source, vcf_file, project_accession)
+    clustering_artifact_path = get_clustering_artifact(clustering_artifact, private_config_file)
+    properties_path = create_properties_file(source, vcf_file, project_accession, assembly_accession,
+                                             private_config_file, private_config_xml_file, profile, output_directory)
+    command = generate_bsub_command(assembly_accession, properties_path, clustering_artifact_path, automation_timestamp)
+    if not only_printing:
+        run_command_with_output('Run clustering command', command, return_process_output=True)
+
+
+def preliminary_check(source, vcf_file, project_accession):
```

Comment:
isn't this exactly the same as in create_clustering_properties.py ?

Reply:
yes, it was. It has changed and is not duplicated anymore